runtime/bundles/org.eclipse.e4.core.di/src/org/eclipse/e4/core/internal/di/FieldRequestor.java

```diff
@@ -44,9 +44,7 @@ private boolean setField(Field field, Object value) throws InjectionException {
 		Object userObject = getRequestingObject();
 		if (userObject == null)
 			return false;
-		if (!field.isAccessible()) {
-			field.setAccessible(true);
-		}
+		field.trySetAccessible();
```

Comment:
Why don't you just replace `isAccessible()` with `canAccess()`? 

Using `trySetAccessible()` changes the logic: now there will no exception thrown if the security manager wouldn't allow to modify the method's accessibility (it did before).

How about:
* Replacing `isAccessible()` wit `canAccess()`
* Putting the `if (!canAccess())` inside the `try-catch` block too
* Catching also `InaccessibleObjectException` and `SecurityException` (they can be thrown by `setAccessible(boolean)`) so they are also rethrown inside the `InjectionException`. 

?

Reply:
Good point: `trySetAccessible(boolean)` also checks the permissions 👍 

Thank you!